modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-admin/dataSetsPermissions.spec.ts

```diff
@@ -0,0 +1,734 @@
+		await setupUserRoleAndLoginAsUser({
+			apiHelpers,
+			dataSetResourcePermissions: [
+				{
+					actions: ['Delete', 'View'],
+					name: 'Data Set',
+				},
+			],
+			page,
+			roleDefinePermissionsPage,
+			rolePage,
+			rolesPage,
+		});
+	});
+
+	await test.step('Go to Data Sets', async () => {
+		await dataSetsPage.goto({checkTabVisibility: false});
+	});
+
+	await test.step('Open actions dropdown', async () => {
+		await openActionsDropdown({
+			page,
+			text: blogPostsDataSetConfig.name,
+		});
+	});
+
+	await test.step('Check that "Delete" is visible', async () => {
+		await expect(dataSetsPage.dataSetDeleteAction).toBeVisible();
+	});
+});
+
+test('Check "Edit" permission', async ({
+	apiHelpers,
+	dataSetManagerApiHelpers,
+	dataSetsPage,
+	page,
+	roleDefinePermissionsPage,
+	rolePage,
+	rolesPage,
+}) => {
+	const blogPostDataSetERC = getRandomString();
+
+	await test.step('Create a data set', async () => {
+		createdDataSetERCs.push(blogPostDataSetERC);
+
+		await dataSetManagerApiHelpers.createDataSet({
+			...blogPostsDataSetConfig,
+			erc: blogPostDataSetERC,
+			label: blogPostsDataSetConfig.name,
+		});
+	});
+
+	await test.step('Setup user role and login as user', async () => {
+		await setupUserRoleAndLoginAsUser({
+			apiHelpers,
+			dataSetResourcePermissions: [
+				{
+					actions: ['View'],
+					name: 'Data Set',
+				},
+			],
+			page,
+			roleDefinePermissionsPage,
+			rolePage,
+			rolesPage,
+		});
+	});
+
+	await test.step('Go to Data Sets', async () => {
+		await dataSetsPage.goto({checkTabVisibility: false});
+	});
+
+	await test.step('Open actions dropdown', async () => {
+		await openActionsDropdown({
+			page,
+			text: blogPostsDataSetConfig.name,
+		});
+	});
+
+	await test.step('Check that "Edit" is not visible', async () => {
+		await expect(dataSetsPage.dataSetEditAction).not.toBeVisible();
+	});
+
+	await test.step('Check that the user can not enter to Data Set details pages', async () => {
+		const dataSetRows = page
+			.locator('.data-set-content-wrapper .dnd-tbody .dnd-tr')
+			.filter({
+				hasText: blogPostsDataSetConfig.name,
+			});
+
+		await dataSetRows
+			.first()
+			.getByText(blogPostsDataSetConfig.name)
+			.first()
+			.click();
+
+		await expect(
+			page.getByRole('button', {name: 'Details'})
+		).not.toBeVisible();
+
+		await page.goto(
+			`${liferayConfig.environment.baseUrl}/group/guest/~/control_panel/manage?p_p_id=com_liferay_frontend_data_set_admin_web_internal_portlet_FDSAdminPortlet&p_p_lifecycle=0&_com_liferay_frontend_data_set_admin_web_internal_portlet_FDSAdminPortlet_mvcRenderCommandName=%2Ffrontend_data_set_admin%2Fedit_data_set&_com_liferay_frontend_data_set_admin_web_internal_portlet_FDSAdminPortlet_dataSetERC=${blogPostDataSetERC}&_com_liferay_frontend_data_set_admin_web_internal_portlet_FDSAdminPortlet_dataSetLabel=${blogPostsDataSetConfig.name}`
+		);
+		await waitForAlert(page, 'Error:Your request failed to complete.', {
+			type: 'danger',
+		});
```

Comment:
Having link hardcoded here is very fragile. Something in there can easily change, we can rename a param or action name, and this test will continue to correctly fail, but for the wrong reasons.

I think that just checking UI enough, that button is not visible. But if we really really need to test this, what we can do it place URL somewhere hidden on the page, like `data-detailsurl` attribute of some element. Then read it in test and try to navigate to it.

Reply:
I cannot find an easy way to add a `data-*`.
The edition url is generated from the click edit callback, triggered from FDS if there is an available action and the cell rendering logic is hidden in the FrontendDataSet component.